### ISSUE TITLE
In pub_client.c ... create parameter to increase size of input buffer beyond 1024

### ISSUE THREAD

**Issue by akobelan** [eclipse/mosquitto#137]
In pub_client.c, messages arriving via stdin are silently truncated to 1024 characters. There is no indication in the documentation of of this limit. Sending JSON strings in excess of 1024 characters causes them to get truncated and arrive as invalid JSON. 

Input messages in excess of 1024 characters should trigger an error message, and not fail silently. Alternately, it should be possible to specify the maximum input message length via an option "-B".

Lines in question: 135 in pub_client.c

	while(!feof(stdin)){
		rlen = fread(buf, 1, 1024, stdin);
		aux_message = realloc(message, pos+rlen);

**Comment by akobelan** [comment]
Sorry. Just saw that this was fixed.

**Comment by ralight** [comment]
No problem, thanks for your input anyway.

**Comment by akobelan** [comment]
Is there any way you could officially increase the 1024 limit to 4096, or have that number overridden
with a command line option, e.g. "-B 4096"?

I'm hitting the limit reading data coming in from STDIN and noisy failure if great, but I need more than 1024 I'm out of options.

**Comment by ralight** [comment]
As you noted back in March, this is no longer an issue whether you're using `-l` or `-s` to read from stdin. The buffer expands automatically in both cases so there is no limit.

**Comment by akobelan** [comment]
I read my comments from March, and then noted a reply to another poster you made stating that the problem was fixed in 1.4.5 (from 1.4.4). I looked at the code and saw you made changes yet I came across the problem once again when using Node-Red. (Messages truncated to 1022 bytes).

Sorry for possibly misattributing the problem to you, Roger. What is happening is I read in data using mosquitto_sub, act upon it in an awk script, and then output the data with mosquitto_pub. Node-Red has an MQTT node that reads in the string via MQTT and outputs it to a web socket, via another node. When I look at the node in Node-Red debug messages, it is truncated to 1022 bytes from possibly a 4k long string. While the problem may be buried in Node-Red, I thought the problem could have been a manifestation of the original issue. 

I'll investigate further and feel reassured that your position is that the problem is not within mosquitto. 
As always, the work you do is much appreciated.

**Comment by akobelan** [comment]
I'm back. Here is a conclusive test. Fill a file with any number of characters in excess of 1024 bytes on a single line. Many ways to do this depending on your OS. I used a 4k+ file of characters on a single
line.

Then ...

```
$ # Let the following line run for a few seconds. Not too long or you will fill up your disk. :)
$ seq -f%04096.0f 1000000 > file_you_generated
```

Launch a background process to capture the topic we will be publishing to.

```
$ mosquitto_sub -t local/test > file_you_generated.out &
```

Publish the big file of 4K long lines.

```
$ cat file_you_generated | mosquitto_pub -l -t local/test
```

when the "pub" is done, bring the "sub" into the foreground and kill the process. You will be left
with two files that you can compare.

Generate a diff ...

```
$ diff file_you_generated file_you_generated.out
```

The output will reveal that the 1023'd character is "\n", the newline. For all the "1022" chunks that are possible in the file. I'm 100% certain there is "bug" in your otherwise excellent code that chunkifys the input  to "1022" byte lines.

Which is why my Node-Red nodes process it accordingly as 1022 byte lines. I humbly submit that the
file should come out exactly as it went in.

**Comment by ralight** [comment]
Hi Alan,

Thanks for the test case, it's always appreciated. You should rest assured that I did do my own testing before I replied yesterday.

I've carried out your test (but using 1000 lines instead of 1000000) and recorded it as an asciicast[1](https://asciinema.org/a/aex522ey5s7ehig0r3s96bfdv). You can see there that it works fine for me using version 1.4.10.

Have you perhaps got an errant version of mosquitto_pub lingering somewhere on your system?

**Comment by akobelan** [comment]
After some sage advice from Roger to make sure I was running the latest code base, or at least something later than 1.4.4 when the fix I requested was applied ... I discovered that I was running 1.4.4 on one of a dozen machines causing me grief.

Thanks, Roger for your patience. 

I was using the package manager on a 14.04 machine that was running 1.4.3 as the latest release at the time that I manually installed 1.4.4 on from sources. As I transitioned to 15.10 and assumed that my
packages were being kept up to date, they were, all but my 1.4.4 manual install. 

With 1.4.10 now running happily on my machine I can report that everything works as expected.
